### DISCUSSION TITLE
Bug? Calendar sessions seems not cleared properly when unsubscribed

### DISCUSSION THREAD

**Discussion by arminlinder** [kewisch/gdata-provider#325] [Support]
I use latest Thunderbird (91.3.2) on Windows 10, and the "Provider for Google Calendar" add-on 91.0.2

My situation is as follows: I had created, many years ago, a GMail account for testing, say mytest@gmail.com. Later I created another account with my real name, named myname@gmail.com. Somehow those accounts got kinda "merged" at Google, the result was that mytest@gmail.com was the "primary" address of the account, and myname@gmail.com was secondary. It seems that those assignments cannot be changed at Google, and the primary account cannot be deleted or renamed --> no way to get rid of mytest@gmail.com. Since I didn't use GMail as my mail provider anyway, I didn't care much until I started using the Google calendar, now I do I feel I need to resolve this situation because I get troubles synchronizing appointments with other calendar users.

So recently I removed myname@gmail.com from mytest@gmail.com and thereafter was able to create a new Google account named myname@gmail.com, and like expected I got a seperate (empty) calendar --> check. Swapping my mobile phone to the new account and importing the old calendar entries from a backup into the new calendar worked too --> check. Right now I have the calendar working like intended on my mobile phone and on GMail.com through the Browser.

The only system which does not get along with the situation is Thunderbird. I deleted the old calendar (myname@gmail.com --> linked to --> mytest@gmail.com), and tried to attach the new calendar. I noticed that, though I removed the old calendar, myname@gmail.com is still listed as "existing session" in the "Create New Calendar" dialog. Regardless of wether I pick that, or follow the "add session" path by entering myname@gmail.com manually, I always and without any chance to change something do end up in the mytest@gmail.com calendar.

My assumption is that though I removed the old calendar there is some persistent local data left over so myname@gmail.com still gets automatically "redirected" to mytest@gmail.com during the addition of a new Google calendar.

How can I resolve this?

Thnx!

**Comment by arminlinder** [comment]
Still stuck with this nagging problem. Meanwhile I have the following additional infos: the problem goes away, if I clear the Thunderbird\profiles folder from the %appdata% directory. Unfortunately this will also kill my local mail archives, so this is not a good option. It does, however, bolster the theory that the problem is caused by persistent settings. I found several occurrances  of mytest@gmail.com in several .json files after deleting the calendar, and even tried to remove them by hand, but had no luck. The old calendar keeps comping back as soon as I try to attach the new one, like described above.

**Comment by kewisch** [comment]
Hey Armin, if you haven't resolved this already, can you check in the password manager? There should be some entries around gdata that you could delete.

**Comment by arminlinder** [comment]
Hi and thanks you for your response, I could not resolve the issue, but found a working solution by using the CALDAV connector. If my issue is of general interest and needs more investigation, I can always try to reproduce the old situation and check if your hint would have helped. Shall I?

Armin.

**Comment by markschultz** [comment]
I'm having a similar issue. My work email migrated to a new google workspace w/ a totally new set of calendars and a new google password. I don't seem to be able to fully remove my existing google account from this addon so that i can re-add it from scratch. when i click add calendar and select I want to use a google account I type my account (which is the same as the old address) and I am presented with what must be some cached list of calendars from the old google workspace instead of a prompt to allow access to the new workspace email.

I've tried uninstalling the addon and reinstalling it completely including removing all subscribed calendars. I don't see a way to fully scrub out this old account.
